data-plane/src/acme/cert.rs

```diff
@@ -241,57 +273,83 @@ impl AcmeCertificateRetreiver {
-                        tokio::spawn(async move {
-                            let _ = self_clone
-                                .create_certificate_and_persist_with_lock(
-                                    key_clone,
-                                    &enclave_context_clone,
-                                    attempts + 1,
-                                )
-                                .await;
-                        });
-                        Some(decrypted_certificate.to_certified_key(x509s, key)).transpose()
-                    }
-                }
-                RenewalStrategy::SyncRenewal => {
-                    log::info!("[ACME] Certificate expires in the coming week. Should be renewed synchronously");
-                    Ok(None)
-                }
-                RenewalStrategy::NoRenewal => {
-                    log::info!("[ACME] Certificate expires in more than a month. No need to renew");
-                    Some(decrypted_certificate.to_certified_key(x509s, key)).transpose()
+        self.schedule_certificate_renewal(
+            time_till_renewal_required,
+            key.clone(),
+            enclave_context.clone(),
+        );
+
+        Some(decrypted_certificate.to_certified_key(x509s, key)).transpose()
+    }
+
+    fn schedule_certificate_renewal(
+        &self,
+        time_till_renewal: Duration,
+        key: PKey<Private>,
+        enclave_context: EnclaveContext,
+    ) {
+        let self_clone = self.clone();
+
+        tokio::spawn(async move {
+            tokio::time::sleep(time_till_renewal).await;
+
+            //Retry certificate renewal if it fails
+            let retries = 3;
+            for _ in 0..retries {
+                if let Err(e) = self_clone
+                    .renew_certificate_and_update_store(key.clone(), &enclave_context)
+                    .await
+                {
+                    log::error!("[ACME] Error renewing certificate: {:?}", e);
+                } else {
+                    break;
                 }
             }
-        } else {
-            log::info!("[ACME] Certificate not found in storage");
-            Ok(None)
+        });
+    }
+
+    async fn renew_certificate_and_update_store(
+        &self,
+        key: PKey<Private>,
+        enclave_context: &EnclaveContext,
+    ) -> Result<(), AcmeError> {
+        let order_lock_maybe = Self::get_order_lock().await?;
+        let attempts = order_lock_maybe
+            .as_ref()
+            .map_or(0, |lock| lock.number_of_attempts().unwrap_or(0));
+
+        if Self::order_lock_exists_and_is_valid(&order_lock_maybe)? {
+            log::info!("[ACME] Lock is valid, waiting for Certificate to be created by other instance or waiting for lock to expire");
+            // If the lock is valid, do nothing and wait for the certificate to be renewed by another instance or for the lock to expire.
+            return Ok(());
```

Comment:
In this case - the scheduled renewal is only created on startup, so if a refresh were to fail, and the other instance was to drop off having seen the lock, would we be able to retry without a restart taking place? 

May be good to add a backoff in the renewal loop to spread the attempts out a bit.